### ISSUE TITLE
IDEX Support for duplicate and mirror mode

### ISSUE THREAD

**Issue by mikepaxton** [OllisGit/OctoPrint-FilamentManager#83]
Add support for IDEX printer support in duplicate and mirror mode.

When printing in either Duplicate or Mirror mode on an IDEX printer the second print head (tool 1) does not subtract the filament used. Currently I have to remember to manually subtract the filament used on tool 1 when printing in either mode.

Provide a means of indicating that the second tool head is being used for current print. Maybe a checkbox next to tool 1 indicating IDEX mode

**Comment by OllisGit** [comment]
Hi @mikepaxton,

how does a gcode file looks like. Do you have sample file?

**Comment by mikepaxton** [comment]
attached is a file that is setup for Duplication Mode.

The "M605 S2 X150 ; Duplicate Mode" is what puts it into duplication mode.
[Eye(DupMode).gcode.zip](https://github.com/OllisGit/OctoPrint-FilamentManager/files/7627407/Eye.DupMode.gcode.zip)

**Comment by mikepaxton** [comment]
Hi @OllisGit ,

Does the above attached .gcode file give you the information you are looking for?  If not let me know what I can do to help you on this.  Would be great for a lot of us who use IDEX printers to have the second color get updated in the filament database

**Comment by OllisGit** [comment]
Hi @mikepaxton,

I looked into the implementation and found the issue. 
The sourcecode to track the filament movement was almost copied from OctoPrint itself, so the IDEX-Mode is already recognized. 
But tracking is activated after the first "movement" of the second tool.

So, I just changed your gcode: set the T1 to a zero position and continued with T0. Just activating via M605 is not enough.

```
T1
G92 E0.0000
T0
G92 E0.0000
```  

And now the filament of both tools is tracked:
![image](https://user-images.githubusercontent.com/24205767/149226094-183fce67-8900-4c46-b547-9214af9f09d9.png)

Solves this your problem?
